ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/MicrosoftSentinelForm.tsx

```diff
@@ -0,0 +1,529 @@
+                clientCert: yup.string().trim(),
+                privateKey: yup.string().trim(),
+            }),
+        }),
+    }),
+    updatePassword: yup.bool(),
+});
+
+export const defaultValues: MicrosoftSentinelFormValues = {
+    notifier: {
+        id: '',
+        name: '',
+        microsoftSentinel: {
+            logIngestionEndpoint: '',
+            directoryTenantId: '',
+            applicationClientId: '',
+            clientCertAuthConfig: {
+                clientCert: '',
+                privateKey: '',
+            },
+            secret: '',
+            alertDcrConfig: {
+                dataCollectionRuleId: '',
+                streamName: '',
+                enabled: false,
+            },
+            auditLogDcrConfig: {
+                dataCollectionRuleId: '',
+                streamName: '',
+                enabled: false,
+            },
+        },
+        labelDefault: '',
+        labelKey: '',
+        uiEndpoint: window.location.origin,
+        type: 'microsoftSentinel',
+    },
+    updatePassword: true,
+};
+
+function MicrosoftSentinelForm({
+    initialValues = null,
+    isEditable = false,
+}: IntegrationFormProps<MicrosoftSentinel>): ReactElement {
+    const formInitialValues = { ...defaultValues, ...initialValues };
+    if (initialValues) {
+        formInitialValues.notifier = {
+            ...formInitialValues.notifier,
+            ...initialValues,
+        };
+        // We want to clear the password because backend returns '******' to represent that there
+        // are currently stored credentials
+        formInitialValues.notifier.microsoftSentinel.secret = '';
+        // Don't assume user wants to change password; that has caused confusing UX.
+        formInitialValues.updatePassword = false;
+    }
+    const {
+        values,
+        touched,
+        errors,
+        dirty,
+        isValid,
+        setFieldValue,
+        handleBlur,
+        isSubmitting,
+        isTesting,
+        onSave,
+        onTest,
+        onCancel,
+        message,
+    } = useIntegrationForm<MicrosoftSentinelFormValues>({
+        initialValues: formInitialValues,
+        validationSchema,
+    });
+    const { isCreating } = usePageState();
+    const [selectedAuthMethod, setSelectedAuthMethod] = useState(
+        values.notifier.microsoftSentinel.clientCertAuthConfig.clientCert
+            ? 'use-client-cert'
+            : 'use-secret'
+    );
+
+    function onChange(value, event) {
+        return setFieldValue(event.target.id, value);
+    }
+
+    function onUpdateAuthMethod(event) {
+        const { id } = event.currentTarget;
+
+        setSelectedAuthMethod(id);
+    }
+
+    function onUpdateCredentialsChange(value, event) {
+        setFieldValue('notifier.microsoftSentinel.secret', '');
+        setFieldValue('notifier.microsoftSentinel.clientCertAuthConfig.privateKey', '');
+        return setFieldValue(event.target.id, value);
+    }
```

Comment:
Could we add explicit types to these function parameters? Without them TypeScript is inferring them to be `any` which could cause problems in the future (and adds more places we'd have to change if we ever want to enable `noImplicitAny`...)